src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -2536,6 +2536,21 @@ void AuraEffect::HandleAuraModSkill(AuraApplication const* aurApp, uint8 mode, b
     target->ModifySkillBonus(prot, (apply ? points : -points), GetAuraType() == SPELL_AURA_MOD_SKILL_TALENT);
 }
 
+void AuraEffect::HandleAuraAllowTalentSwapping(AuraApplication const* aurApp, uint8 mode, bool apply) const
+{
+    if (!(mode & (AURA_EFFECT_HANDLE_REAL)))
+        return;
+
+    Player* target = aurApp->GetTarget()->ToPlayer();
+    if (!target)
+        return;
+
+    if (apply)
+        target->AddUnitFlag2(UNIT_FLAG2_ALLOW_CHANGING_TALENTS);
+    else
```

Comment:
As with all aura types that apply a flag, it needs this kind of check to properly handle stacking
```
    if (!apply && target->HasAuraType(GetAuraType()))
        return;
```